task/common/values.go

```diff
@@ -41,7 +41,15 @@ type Event struct {
 	Code        string
 	Description []string
 }
-type Storage struct{}
+type RemoteStorage struct {
+	// Container stores the id of the container to be used.
+	Container string
+	// Path stores the subdirectory inside the container.
+	Path string
+	// Config stores provider-specific configuration keys for accessing the pre-allocated
+	// storage container.
+	Config map[string]string
```

Comment:
Maybe too much flexibility? I'd assume that no special options are needed to connect and, if they were, users could just provide a full [`rclone` connection string](https://rclone.org/docs/#connection-strings), removing the maintenance burden on our side.

Reply:
It depends on what we're aiming at. Rclone connection strings hardly make it easy to configure.

The difference is between
```
  storage {    
    container = "tpi-test"
    container_path = "directory"
    container_opts {
      az_storage_account = "storage-account"
      az_storage_account_key = "storage-account-key"
    }
  }
```
and
```
  storage {
    container = ":azureblob,account='storage-account',key='storage-account-key':tpi-test/directory"
  }
```

I'm not sure the latter is easier, especially if we consider the target users of tpi.

Reply:
More precisely, the difference would be between

```
  storage {    
    container = "tpi-test"
    container_path = "directory"
    container_opts {
      az_storage_account = "storage-account"
      az_storage_account_key = "storage-account-key"
    }
  }
```

and

```
  storage = ":azureblob,account='storage-account',key='storage-account-key':tpi-test/directory"
```

Reply:
None of the variants sound particularly easy to me, although the latter is already well docuemnted by `rclone` and, despite being a lengthy string, is fairly readable to my mind.

Reply:
> _I'm not sure the latter is easier, especially if we consider the target users_

That's a double-edged reason: if our target users are on their own, they probably won't need to specify a custom storage and, if they do, it probably won't require any options beyond the name and path (prefix). 😸 

Note that credentials usually match (or can be retrieved by using) the main credentials used to create resources; see https://github.com/iterative/terraform-provider-iterative/pull/651#discussion_r962436130 for details.

And those users who really need custom options on their connection strings are probably part of a team with opinions, and people having those opinions may as well read `rclone` documentation. 😈 Worst-case scenario, we can provide an interactive `rclone` string builder as part of our documentation.